kernel/kernel-api/src/main/java/io/delta/kernel/internal/types/DataTypeJsonSerDe.java

```diff
@@ -211,14 +221,23 @@ private static StructType parseStructType(JsonNode json) {
   private static StructField parseStructField(JsonNode json) {
     Preconditions.checkArgument(json.isObject(), "Expected JSON object for struct field");
     String name = getStringField(json, "name");
-    DataType type = parseDataType(getNonNullField(json, "type"));
+    FieldMetadata metadata = parseFieldMetadata(json.get("metadata"), false);
+    DataType type =
+        parseDataType(getNonNullField(json, "type"), name, getCollationsMap(json.get("metadata")));
```

Comment:
why not just use the parsed `metadata` from above and pass to `getCollationsMap`?

Reply:
FieldMetadata class doesn't have any remove methods, so I tought that it should stay the same (we won't have collation info in parsed struct field)